### ISSUE TITLE
java.lang.IllegalArgumentException: Profile array must not be null

### ISSUE THREAD

**Issue by Euklios** [hank-cp/sbp#6]
Hello there, quite new to pf4j and sbp. So I'm probably just missing something.

I'm trying to move an existing pf4j application to sbp. However, when starting the plugin-container, I receive an IllegalArgumentException from spring. According to the stack-trace, the responsible spring boot method is directly called by sbp.

I've tried to specify spring-profiles int the container, as well as in the loaded test-plugin, however, that seems not to be the profiles, spb is using.
Is there any way to specify the used profiles / make that array not null?

spring-boot-starter-parent: 2.2.6.RELEASE
sbp-core: 0.1.3
java: 11

Logs:
[sbp.log](https://github.com/hank-cp/sbp/files/4611844/sbp.log)

**Comment by Euklios** [comment]
Could be related: My shared module has the scope provided. Therefore, it will not be included in the bundled plugin.
This also includes sbp-core.

I'm not sure dough if this matters since pf4j without sbp could be loaded in this setup.

**Comment by hank-cp** [comment]
Everny plugin has its onw `ApplicationContext`, as well as `enviornment` and `profiles`. The `profiles` you are missing here is for plugin only, but not for the main app. So if you were trying to tell main app container the `profiles` it wouldn't take effect to the plugins.

The missing `profiles` is declare [here](https://github.com/hank-cp/sbp/blob/45d2ddeefd8996474b98ac5f3056495292a899fc/sbp-spring-boot-starter/src/main/java/org/laxture/sbp/spring/boot/SbpProperties.java#L56). Document is a bit outdate and I will update it later. 

But the strange thing is, `SbpProperties.profiles` has default value and it shouldn't be null. Could you have a check that have you ever set this properties at somewhere?

Also, please have a test with latest version `0.1.4` or `-SNAPSHOOT`

**Comment by hank-cp** [comment]
The profiles related testcase is [here](https://github.com/hank-cp/sbp/blob/45d2ddeefd8996474b98ac5f3056495292a899fc/demo-app/src/test/java/demo/sbp/app/PluginIntegrationTest.java#L441), hope it helps.

**Comment by Euklios** [comment]
I did not write it, but I'm aware that plugins have their own `ApplicationContext`. That's why I did my experiments within the plugins, as well as in the container application. Only exceptions are places where I can't specify profiles for one of those. For example, CLI-parameters won't work for the plugin when loaded using the container.

I've noticed the default value specified in SbpProperties.profiles, and I'd agree with you, that should never be null unless explicitly set to null by the corresponding setter (and this doesn't happen according to my breakpoint).

I have specified `spring.sbp.runtimeMode`, `spring.sbp.enabled` and `spring.sbp.classes-directories` within my application.properties file in the container application, but I've never touched `spring.sbp.profiles`.

Just did some testing and added `spring.sbp.profiles=plugins` to both `application.properties` files (container and plugin) and the `plugin.properties` file (probably useless, as far as I understand we are talking about spring-profiles). However, this did not affect the exception at all.

I also tried to use `bootstrap.setAdditionalProfiles("plugins")` when creating the `SpringBootstrap` object, to no avail.

Btw. I did update to 0.1.4 before testing all of this. However, I was unable to find a SNAPSHOT artifact on maven-central.

I ve attached my current *.properties files, with names changed to satisfy GitHub-restrictions.

[container_application.properties.txt](https://github.com/hank-cp/sbp/files/4618292/container_application.properties.txt)
[plugin.properties.txt](https://github.com/hank-cp/sbp/files/4618281/plugin.properties.txt)
[plugin_application.properties.txt](https://github.com/hank-cp/sbp/files/4618282/plugin_application.properties.txt)

**Comment by hank-cp** [comment]
I guess this line cause the problem:
```
spring.sbp.classes-directories = "out/production/classes, out/production/resources"
```

should be:
```
spring.sbp.classes-directories=out/production/classes,out/production/resources
```

BTW, `plugin.properties` is not where `spring.sbp.profiles=plugin` to go, it should have been placed at `plugin_application.properties`

And, if you want to use SNAPSHOT, just chnage version part to `-SNAPSHOT`. It's in maven central's SNAPSHOP repository, but the website is just showing the RELEASE repository.

**Comment by Euklios** [comment]
Sorry for the delay, quite a lot going on in my life right now.

I've changed `spring.sbp.classes-directories` to the format you've provided and also noticed, that these values don't make sense in my environment. My output is stored in `target/classes`, relative to the modules directory, so I've changed it to that folder.
Sadly this did not change anything. 

Afterward, I was working on a minimal viable project to reproduce my issue, since I thought that it would probably be more comfortable for us both to understand what was going on. During this, I noticed that my project was missing the `ApplicationContextAware` Bean. That did fix the issue / moved me to the next one.

After some trying, rebuilding, a night of sleep, and removing some libraries, I finally managed to get it somewhat working.
I still have to implement DB-migrations, and I hope that I manage to switch back from flyway to liquibase (We have to support multiple different database-engines, extremely tedious to do in flyway).

You've helped me quite a lot. Thank a lot for you're aid.
